src/platform/tests/TestThreadStackMgr.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <assert.h>
+#include <dbus/dbus.h>
+#include <memory>
+
+#include "platform/internal/CHIPDeviceLayerInternal.h"
+
+#include "platform/PlatformManager.h"
+#include "platform/ThreadStackManager.h"
+
+#if CHIP_DEVICE_LAYER_TARGET == LINUX
+#include <thread>
+
+struct DBusConnectionDeleter
+{
+    void operator()(DBusConnection * aConnection) { dbus_connection_unref(aConnection); }
+};
+
+using UniqueDBusConnection = std::unique_ptr<DBusConnection, DBusConnectionDeleter>;
+#endif
+
+void EventHandler(const chip::DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
+{
+    (void) arg;
+    if (event->Type == chip::DeviceLayer::DeviceEventType::kThreadConnectivityChange)
+    {
+        if (event->ThreadConnectivityChange.Result == chip::DeviceLayer::ConnectivityChange::kConnectivity_Established)
+        {
+            exit(0);
+        }
+    }
+}
+
+int TestThreadStackManager(void)
+{
+#if CHIP_DEVICE_LAYER_TARGET == LINUX
```

Comment:
Can these platform differences be captured inside `ThreadStackManagerImpl` constructor?  Ideally, the platform abstraction layer would remove the need for any ifdef in application of test code using that abstraction API.

Reply:
I'd like to view the dbus initialization and IO loop as a temporary fix before the dbus module in Linux platform manager is ready. Once it's ready we should handle all the dbus things there rather than in the Thread manager.

Reply:
Is there any issue with moving this code into either `impl.InitThreadStack()` or `impl.StartThreadTask()`?